pantheon-bundle/frontend/src/app/product.tsx

```diff
@@ -125,8 +145,52 @@ class Product extends Component<any, IState> {
         }
     }
 
-    private saveProduct = () => {
-        if (this.state.productName === ""|| this.state.productUrlFragment === "") {
+    private handleTextInputChange = versionName => {
+        this.setState({ versionName })
+    }
+
+    private handleUrlInputChange = versionUrlFragment => {
+        if (/^[-.\w]+$/.test(versionUrlFragment)) {
+            this.setState({ versionUrlFragment, isUrlFragmentValid: true })
+        } else {
+            this.setState({ isUrlFragmentValid: false})
+        }
+    }
+
+    private saveVersion = () => {
+        // confirming form state is valid
+        if (this.state.versionName === "" || this.state.productUrlFragment === "") {
```

Comment:
line 162 does not check if users can enter an empty string in Version Name field. It would be better if we do something like this:
if (this.state.versionName.trim() === "" || this.state.versionUrlFragment === "")